posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -0,0 +1,648 @@
+        }
+
+        @Suppress("UNCHECKED_CAST")
+        val groups = memoryPreferences.getValue("\$groups") as? Map<String, Any>
+        val newGroups = mutableMapOf<String, Any>()
+        var reloadFeatureFlags = false
+
+        groups?.let {
+            val currentKey = it[type]
+
+            if (key != currentKey) {
+                reloadFeatureFlags = true
+            }
+
+            newGroups.putAll(it)
+        }
+        newGroups[type] = key
+
+        capture("\$groupidentify", properties = props)
+
+        memoryPreferences.setValue("\$groups", newGroups)
+
+        if (reloadFeatureFlags) {
+            loadFeatureFlagsRequest(null)
+        }
+    }
+
+    public override fun reloadFeatureFlags(onFeatureFlags: PostHogOnFeatureFlags?) {
+        if (!isEnabled()) {
+            return
+        }
+        loadFeatureFlagsRequest(onFeatureFlags)
+    }
+
+    private fun loadFeatureFlagsRequest(onFeatureFlags: PostHogOnFeatureFlags?) {
+        val props = mutableMapOf<String, Any>()
+        props["\$anon_distinct_id"] = anonymousId
+        props["distinct_id"] = distinctId
+
+        @Suppress("UNCHECKED_CAST")
+        val groups = memoryPreferences.getValue("\$groups") as? Map<String, Any>
+
+        featureFlags?.loadFeatureFlags(distinctId, anonymousId, groups, onFeatureFlags)
+    }
+
+    public override fun isFeatureEnabled(key: String, defaultValue: Boolean): Boolean {
+        if (!isEnabled()) {
+            return defaultValue
+        }
+        return featureFlags?.isFeatureEnabled(key, defaultValue) ?: defaultValue
+    }
+
+    public override fun getFeatureFlag(key: String, defaultValue: Any?): Any? {
+        if (!isEnabled()) {
+            return defaultValue
+        }
+        val flag = featureFlags?.getFeatureFlag(key, defaultValue) ?: defaultValue
+
+        if (config?.sendFeatureFlagEvent == true) {
+            val props = mutableMapOf<String, Any>()
+            props["\$feature_flag"] = key
+            flag?.let {
+                props["\$feature_flag_response"] = it
+            }
+
+            capture("\$feature_flag_called", properties = props)
+        }
+
+        return flag
+    }
+
+    public override fun getFeatureFlagPayload(key: String, defaultValue: Any?): Any? {
+        if (!isEnabled()) {
+            return defaultValue
+        }
+        return featureFlags?.getFeatureFlagPayload(key, defaultValue) ?: defaultValue
+    }
+
+    public override fun flush() {
+        if (!isEnabled()) {
+            return
+        }
+        queue?.flush()
+    }
+
+    public override fun reset() {
+        if (!isEnabled()) {
+            return
+        }
+
+        memoryPreferences.clear()
+        // only remove properties, preserve BUILD and VERSION keys in order to to fix over-sending
+        // of 'Application Installed' events and under-sending of 'Application Updated' events
+        config?.cachePreferences?.clear(listOf("build", "version"))
```

Comment:
```suggestion
        config?.cachePreferences?.clear(except = listOf("build", "version"))
```

I'd keep the parameter named here, on first scan I didn't read the comment and assumed this was _only_ clearing build and version.

Reply:
Good point